core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGeneData.java

```diff
@@ -607,96 +683,105 @@ static void importSuppGeneData(File suppGeneFile, String referenceGenomeBuild) t
-                    numLines = FileUtil.getNumLines(lociFile);
-                    System.out.println(" --> total number of lines:  " + numLines);
-                    ProgressMonitor.setMaxValue(numLines);
-                    ImportGeneData.importGeneLength(lociFile, (String)options.valueOf("genome-build"), 
-                                        species, options.has("genes"));
+        try {
+            SpringUtil.initDataSource();
+
+            String description = "Update gene / gene alias tables ";
+
+            // using a real options parser, helps avoid bugs
+            OptionParser parser = new OptionParser();
+            OptionSpec<Void> help = parser.accepts( "help", "print this help info" );
+            parser.accepts( "genes", "ncbi genes file" ).withRequiredArg().describedAs( "ncbi_genes.txt" ).ofType( String.class );
+            parser.accepts( "hgnc", "hgnc genes file" ).withRequiredArg().describedAs( "hgnc_genes.txt" ).ofType( String.class );
+            parser.accepts( "supp-genes", "alternative genes file" ).withRequiredArg().describedAs( "supp-genes.txt" ).ofType( String.class );
+            parser.accepts( "microrna", "microrna file" ).withRequiredArg().describedAs( "microrna.txt" ).ofType( String.class );
+            parser.accepts( "gtf", "gtf file for calculating and storing gene lengths" ).withRequiredArg().describedAs( "gencode.<version>.annotation.gtf" ).ofType( String.class );
+            parser.accepts( "genome-build", "genome build eg GRCh38" ).withRequiredArg().describedAs( "genome build" ).ofType( String.class );
+            parser.accepts( "species", "different kinds of organisms eg. humna").withRequiredArg().describedAs( "species" ).ofType( String.class );
+            String progName = "importGenes";
+            OptionSet options = null;
+            try {
+                options = parser.parse( args );
+            } catch (OptionException e) {
+                throw new UsageException(progName, description, parser,
+                    e.getMessage());
+            }
+
+            if( options.has( help ) ){
+                throw new UsageException(progName, description, parser);
+            }
+
+            ProgressMonitor.setConsoleMode(true);
+
+            if(options.has("genes")) {
+                File geneFile = new File((String) options.valueOf("genes"));
+                System.out.println("Reading gene data from:  " + geneFile.getAbsolutePath());
+                int numLines = FileUtil.getNumLines(geneFile);
+                System.out.println(" --> total number of lines:  " + numLines);
+                ProgressMonitor.setMaxValue(numLines);
+                MySQLbulkLoader.bulkLoadOn();
+                ImportGeneData.importData(geneFile, (String)options.valueOf("genome-build"));
+                MySQLbulkLoader.flushAll(); //Gene and gene_alias should be updated before calculating gene length (gtf)!
+            }
+
+            if(options.has("hgnc")) {
```

Comment:
Maybe this should be something more generic instead of "hgnc" because really there is nothing about the input file that is specific to hgnc.

Reply:
I think hgnc is ok. The general format of the file (number of columns and column content) is appropriate for hgnc processing, and the semantics (such as cytoband expectations) are based on the hgnc content. If we add new possible sources of genes, those may have their own formats and semantics. Possibly we should rename the "-genes" option to "-ncbi" because that file format has semantics specific to the ncbi gene source.